src/pcp.rs

```diff
@@ -412,24 +412,21 @@ struct ProveShimGadget<F: FieldElement> {
 }
 
 impl<F: FieldElement> ProveShimGadget<F> {
-    #[allow(clippy::new_ret_no_self)]
-    fn new(
-        inner: Box<dyn Gadget<F>>,
-        g_calls: usize,
-    ) -> Result<Box<dyn Gadget<F>>, getrandom::Error> {
+    fn new(inner: Box<dyn Gadget<F>>, g_calls: usize) -> Result<Self, PcpError> {
         let mut f_vals = vec![vec![F::zero(); 1 + g_calls]; inner.arity()];
         let mut prng = Prng::new_with_length(f_vals.len())?;
 
-        for wire in &mut f_vals {
+        #[allow(clippy::needless_range_loop)]
+        for wire in 0..f_vals.len() {
             // Choose a random field element as the first point on the wire polynomial.
-            wire[0] = prng.next().unwrap();
+            f_vals[wire][0] = prng.next().unwrap();
         }
 
-        Ok(Box::new(Self {
+        Ok(Self {
             inner,
             f_vals,
             ct: 1,
-        }))
+        })
     }
 }
 
```

Comment:
(GitHub won't let me reach below here.) Several lines down from here we declare two variables, `g_deg` and `g_arity`. We should change these to `gadget_deg` and `gadget_arity` for consistency with related changes elsewhere.

Reply:
With pleasure! I expanded a few more terse variable names while I was in there.